File: Suldin_Vyacheslav/SecondTerm/Task5Folder/Parsers/StormGlassParser.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Newtonsoft.Json.Linq;

namespace Parsers
{
    public class StormGlassParser : JSONParser
    {
        public StormGlassParser()
        {
            parsingParams = "airTemperature,cloudCover,humidity,precipitation,windSpeed,windDirection".Split(",");
        }

        public override IReadOnlyList<string> Parse(JObject json)
        {
            string[] information = new string[8];
            information[0] = "StormGlass";
            if (json["ERROR"] == null)
            {
                int k = 2;
                foreach (string field in parsingParams)
                {
                    information[k] = json["hours"][0][$"{field}"]["noaa"].ToString();
                    if ($"{field}" == "precipitation" && information[k] == "0") information[k] = PrecipitationType.NoPrecip.ToString();
                    k++;
                }

                information[1] = Math.Round((Convert.ToDouble(information[2]) * 9 / 5 + 32), 3).ToString();

                ParsedInfo = information;
            }
            else ParsedInfo = new string[1] { json["ERROR"].ToString() + this.ToString().Split(".")[1].Split("Parser")[0] };
            return ParsedInfo;
        }
    }
}

File: Ilya-Dudnikov/term_2/block_3/task_2/Stormglass/src/main/java/model/JSONParser/StormglassParser.java
package model.JSONParser;

import org.json.simple.JSONArray;
import org.json.simple.JSONObject;
import org.json.simple.parser.ParseException;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;

public class StormglassParser implements JSONParser {
	private static final ArrayList<String> params = new ArrayList<>() {{
		add("airTemperature");
		add("cloudCover");
		add("humidity");
		add("precipitation");
		add("windDirection");
		add("windSpeed");
	}};

	@Override
	public Map<String, Double> parse(String json) throws ParseException {
		JSONObject jsonObject = (JSONObject) new org.json.simple.parser.JSONParser().parse(json);

		var hoursArray = (JSONArray) jsonObject.get("hours");
		JSONObject data = (JSONObject) hoursArray.get(hoursArray.size() - 1);

		return new HashMap<>() {{
			params.forEach((param) -> {
				JSONObject currentElem = (JSONObject) data.get(param);
				if (currentElem == null)
					put(param, null);
				else if (param.equals("airTemperature")) {
					put("airTemperatureC", (Double) currentElem.get("sg"));
					put("airTemperatureF", 1.8 * ((Double) currentElem.get("sg")) + 32);
				} else {
					put(param, (Double) currentElem.get("sg"));
				}
			});
		}};
	}
}
